src/pudl_archiver/depositors/fsspec.py

```diff
@@ -232,7 +243,7 @@ def generate_change(
         remote_path = self.deposition.deposition_path / filename
 
         if remote_path.exists():
-            remote_md5 = compute_md5(remote_path)
+            remote_md5 = self.get_checksum(filename)
             local_md5 = compute_md5(resource.local_path)
             if remote_md5 != local_md5:
```

Comment:
Not sure how this would happen but it might also be good to error out here if either of `remote_md5` or `local_md5` are `None` or really if they're somehow equal but not actually valid md5 checksums. I just mention this because I see `md5_hash` starts out as None in `get_checksum` and `if not filepath.exists()` it never gets set to something else.